contracts/src/KeyManager.sol

```diff
@@ -0,0 +1,278 @@
+        }
+        __Ownable_init(msg.sender);
+        __UUPSUpgradeable_init();
+        manager = initialManager;
+    }
+
+    /**
+     * @notice This function is used to authorize the upgrade of the contract.
+     * @dev Reverts if the caller is not the owner.
+     * @dev Assumes that the new implementation is valid.
+     * @param newImplementation The new implementation.
+     */
+    function _authorizeUpgrade(address newImplementation) internal override onlyOwner {}
+
+    /**
+     * @notice This function is used to set the manager.
+     * @dev Reverts if the manager is the zero address or the same as the current manager.
+     * @dev Reverts if the caller is not the owner.
+     * @dev Assumes that the manager is valid.
+     * @param newManager The new manager.
+     */
+    function setManager(address newManager) external virtual onlyOwner {
+        if (newManager == address(0) || newManager == manager) {
+            revert InvalidAddress();
+        }
+        address oldManager = manager;
+        manager = newManager;
+        emit ManagerChanged(oldManager, newManager);
+    }
+
+    /**
+     * @notice This function is used to set the threshold encryption key.
+     * @dev Reverts if the threshold encryption key is already set.
+     * @dev Reverts if the caller is not the manager.
+     * @dev Assumes that the threshold encryption key is valid.
+     * @param newThresholdEncryptionKey The threshold encryption key.
+     */
+    function setThresholdEncryptionKey(bytes calldata newThresholdEncryptionKey) external virtual onlyManager {
+        if (thresholdEncryptionKey.length > 0) {
+            revert ThresholdEncryptionKeyAlreadySet();
+        }
+        thresholdEncryptionKey = newThresholdEncryptionKey;
+        emit ThresholdEncryptionKeyUpdated(newThresholdEncryptionKey);
+    }
+
+    /**
+     * @notice This function is used to set the next committee.
+     * @dev Reverts if the members array is empty.
+     * @dev Reverts if the effective timestamp is less than the last effective timestamp.
+     * @dev Reverts if the committees mapping is at uint64.max.
+     * @dev Assumes that the committee members are valid.
+     * @param effectiveTimestamp The effective timestamp of the committee.
+     * @param members The committee members.
+     * @return committeeId The id of the new committee.
+     */
+    function setNextCommittee(uint64 effectiveTimestamp, CommitteeMember[] calldata members)
+        external
+        virtual
+        onlyManager
+        returns (uint64 committeeId)
+    {
+        if (members.length == 0) {
+            revert EmptyCommitteeMembers();
+        }
+
+        // ensure the effective timestamp is greater than the last effective timestamp
+        if (nextCommitteeId > 0) {
+            uint64 lastTimestamp = committees[nextCommitteeId - 1].effectiveTimestamp;
+            if (effectiveTimestamp <= lastTimestamp || effectiveTimestamp > lastTimestamp + 10 minutes) {
```

Comment:
```suggestion
            if (effectiveTimestamp <= lastTimestamp) {
```

no need to add the 10 min restriction. each epoch usually takes much longer than 10 min (i thought it was mostly a day or even longer.)